js/src/hooks/useMCProductStatistics.js

```diff
@@ -1,15 +1,60 @@
+/**
+ * External dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
 /**
  * Internal dependencies
  */
-import useAppSelectDispatch from '.~/hooks/useAppSelectDispatch';
+import useAppSelectDispatch from './useAppSelectDispatch';
+import useCountdown from './useCountdown';
+import { useAppDispatch } from '.~/data';
 
 /**
  * Call `useAppSelectDispatch` with `"getMCProductStatistics"`.
+ * If the background process is loading, start a countdown and invalidate  the resolution for getMCProductStatistics and getMCProductFeed.
  *
  * @return {useAppSelectDispatch} Result of useAppSelectDispatch.
  */
 const useMCProductStatistics = () => {
-	return useAppSelectDispatch( 'getMCProductStatistics' );
+	const { second, callCount, startCountdown } = useCountdown();
+	const { invalidateResolutionForStoreSelector } = useAppDispatch();
+	const { data, hasFinishedResolution, invalidateResolution, ...rest } =
+		useAppSelectDispatch( 'getMCProductStatistics' );
+
+	const isLoading = hasFinishedResolution && data?.loading ? true : false;
+	const hasStats = hasFinishedResolution && ! data?.loading ? true : false;
+
+	useEffect( () => {
+		if ( isLoading && second === 0 ) {
+			startCountdown( 30 );
+		}
+
+		if ( isLoading && second === 0 && callCount > 0 ) {
+			invalidateResolution();
+		}
```

Comment:
We could combine this two conditions like:

```suggestion
		if ( isLoading && second === 0 ) {
			 startCountdown( 30 );
			 if ( callCount > 0 ) {
			   invalidateResolution();
			 }
		}
```

Reply:
Also probably might be helpful to add some comments, e.g. `Start counter when...` and `Clear the product statistic data...`

Reply:
I made an update and also renamed the constant from `isLoading` to `isCalculatingStats`. I believe this change clarifies that the loading belongs to the calculation process rather than indicating that the HTTP request is in a "loading" state.